velox/common/memory/SharedArbitrator.h

```diff
@@ -400,21 +417,25 @@ class SharedArbitrator : public memory::MemoryArbitrator {
   int64_t minGrowCapacity(const MemoryPool& pool) const;
 
   // Returns true if 'pool' is under memory arbitration.
-  bool isUnderArbitration(MemoryPool* pool) const;
   bool isUnderArbitrationLocked(MemoryPool* pool) const;
 
   void updateArbitrationRequestStats();
+
   void updateArbitrationFailureStats();
 
   const uint64_t reservedCapacity_;
+  const uint64_t memoryPoolInitialCapacity_;
   const uint64_t memoryPoolReservedCapacity_;
   const uint64_t memoryPoolTransferCapacity_;
   const uint64_t memoryReclaimWaitMs_;
   const bool globalArbitrationEnabled_;
   const bool checkUsageLeak_;
 
+  mutable folly::SharedMutex poolLock_;
+  std::unordered_map<MemoryPool*, std::weak_ptr<MemoryPool>> candidates_;
```

Comment:
Is there a specific reason for using the pointer as the key here instead of `poolName` as in `MemoryManager::pools_`?

Reply:
The memory manager prevents the duplicate pool names while arbitrator just use pointer as index which is more efficient.